### ISSUE TITLE
[Bug] When a view appears it is not accessible via VoiceOver

### ISSUE THREAD

**Issue by boyd-m** [xamarin/Xamarin.Forms#7606]
### Description

If a previously hidden view is toggled to visible, it can not be accessed via VoiceOver

This appears to be linked to https://github.com/xamarin/Xamarin.Forms/issues/6929

### Steps to Reproduce
- Open the app on an iOS device
- Turn voice over on
- Toggle a previously hidden view to visible

### Expected Behavior

- When the view appears, it is reachable by swiping left/right on voice over

### Actual Behavior

- When the view appears, it is not reachable by swiping left/right on voice over

### Basic Information

- Version with issue: 4.2.0.815419
- Last known good version:
- IDE: Visual studio professional (both windows and mac)
- Platform Target Frameworks: 
  - iOS:  12.4 
- Affected Devices:
  - Observed on iPhone 6S

### Reproduction Link

The following sample project contains a demonstration of the issue:
https://github.com/boyd-m/VisibilityA11y
